include/seqan3/range/container/constexpr_string.hpp

```diff
@@ -0,0 +1,430 @@
+    data_type _lit;
+
+    /*!\brief Constructs new constexpr_string by merging two other constexpr_strings.
+     * \param lhs left-hand-side of constexpr_string to merge.
+     * \param rhs right-hand-side of constexpr_string to merge.
+     * \par Exceptions
+     * No-throw guarantee.
+     */
+    template <std::size_t N1>
+    constexpr constexpr_string(constexpr_string<N1> const & lhs,
+                            constexpr_string<N - N1> const & rhs) noexcept
+    {
+        for (unsigned i = 0; i < N1; ++i)
+            _lit[i] = lhs[i];
+        for (unsigned i = N1; i < N + 1; ++i)
+            _lit[i] = rhs[i - N1];
+    }
+
+public:
+
+    /*!\name Public member types
+     * \{
+     */
+    using value_type      = char;
+    using reference       = char &;
+    using const_reference = const char &;
+    using iterator        = std::add_pointer_t<value_type>;
+    using const_iterator  = std::add_pointer_t<value_type const>;
+    using difference_type = typename data_type::difference_type;
+    using size_type       = typename data_type::size_type;
+    //!\}
+
+    /*!\name Constructors and Destructor
+     * \{
+     */
+     //!\brief Default default constructor.
+    constexpr_string() = default;
+    //!\brief Default copy constructor.
+    constexpr_string(constexpr_string const &) = default;
+    //!\brief Default move constructor.
+    constexpr_string(constexpr_string &&) = default;
+    //!\brief Default copy-assignment operator.
+    constexpr_string & operator=(constexpr_string const &) = default;
+    //!\brief Default move-assignment operator.
+    constexpr_string & operator=(constexpr_string &&) = default;
+
+    /*!\brief Construction from literal.
+     * \param lit The literal to construct the string for.
+     * \par Exceptions
+     * No-throw guarantee.
+     */
+    constexpr constexpr_string(const char (&lit)[N + 1]) noexcept
+        : constexpr_string{}
+    {
+        // static_assert(lit[N] == '\0'); TODO(rrahn): Fix me
+        for (unsigned i = 0; i < N + 1; ++i)
+            _lit[i] = lit[i];
+    }
+
+    /*!\brief Construction from char array.
+     * \param src The char array to construct the constexpr_string for.
+     * \par Exceptions
+     * No-throw guarantee.
+     */
+    constexpr constexpr_string(std::array<char, N> const & src) noexcept
+        : constexpr_string{}
+    {
+        // static_assert(src[N-1] != '\0'); TODO(rrahn): Fix me
+        for (unsigned i = 0; i < N; ++i)
+            _lit[i] = src[i];
+        _lit[N] = '\0';
+    }
+
+    /*!\brief Construction from char.
+     * \param c The character to construct the constexpr_string for.
+     * \par Exceptions
+     * No-throw guarantee.
+     */
+    constexpr constexpr_string(const char c) noexcept
+        : constexpr_string{}
+    {
+        _lit[0] = c;
+        _lit[1] = '\0';
+    }
+
+    //!\brief Default destructor.
+    ~constexpr_string() = default;
+    //!\}
+
+    /*!\name Member functions
```

Comment:
"Member functions" is very uninformative, for the other coontainer, I chose to go exactly with the "member categories" also used on cpp-references (possibly even suggested by the standard, I don't know).
see http://en.cppreference.com/w/cpp/container/vector

* Cunstructors, destructor and assignment
* Element access (at front back [] data)
* Iterators (begin end cbegin...)
* Capacity (size capacity reserve)
* Modifier (emplace push_back resize...)

It would be good if we do this consistently, we should also add this to the "Writing documentation" guide.